.bazelrc

```diff
@@ -38,37 +38,26 @@ build:asan-libfuzzer --@rules_fuzzing//fuzzing:cc_engine=@rules_fuzzing//fuzzing
 build:asan-libfuzzer --@rules_fuzzing//fuzzing:cc_engine_instrumentation=libfuzzer
 build:asan-libfuzzer --@rules_fuzzing//fuzzing:cc_engine_sanitizer=asan
 
-# Shared configuration for clang's source-based coverage instrumentation.
-# Bazel seems to support this only partially, thus we have to perform some
-# additional processing. See
-# https://github.com/bazelbuild/bazel/commit/21b5eb627d78c09d47c4de957e9e6b56a9ae6fad
-# and `util/coverage/coverage_off_target.py`.
-build:ot_coverage --repo_env='CC=clang'
-build:ot_coverage --repo_env='BAZEL_USE_LLVM_NATIVE_COVERAGE=1'
-build:ot_coverage --java_runtime_version='remotejdk_11'
-# Docs state that bazel will fail to create coverage information if tests have
-# been cached previously. See
-# https://bazel.build/configure/coverage?hl=en#remote-execution
-coverage:ot_coverage --nocache_test_results
-
-# Configuration for measuring off-target coverage. Enable with
-# `--config=ot_coverage_off_target`.
-build:ot_coverage_off_target --config='ot_coverage'
-build:ot_coverage_off_target --collect_code_coverage
-coverage:ot_coverage_off_target --repo_env='GCOV=/usr/bin/llvm-profdata'
-coverage:ot_coverage_off_target --repo_env='BAZEL_LLVM_COV=/usr/bin/llvm-cov'
-
-# Configuration for measuring on-target coverage. Enable with
-# `--config=ot_coverage_on_target`.
-build:ot_coverage_on_target --config='ot_coverage'
-build:ot_coverage_on_target --platforms="@//toolchain:opentitan_platform"
-build:ot_coverage_on_target --define='measure_coverage_on_target=true'
-# Instrument selectively to limit size overhead when measuring on-target coverage.
-# Note: We have to disable optimizations until the corresponding item in #16761 is
-# resolved.
-build:ot_coverage_on_target --per_file_copt='//sw/device/silicon_creator[/:].*,//sw/device/lib/base:.*@-fprofile-instr-generate,-fcoverage-mapping,-O0'
-# Needed to be able to build host binaries while collecting coverage.
-coverage:ot_coverage_on_target --platforms=""
+# Configuration for clang's source-based coverage instrumentation.
+coverage:ot_coverage --repo_env='CC=clang'
+coverage:ot_coverage --repo_env='BAZEL_USE_LLVM_NATIVE_COVERAGE=1'
+coverage:ot_coverage --java_runtime_version='remotejdk_11'
+
+# Set coverage mode indicators
+coverage:ot_coverage --define='ot_coverage_enabled=true'
+coverage:ot_coverage --@rules_rust//:extra_rustc_flag='--cfg=feature="ot_coverage_enabled"'
+coverage:ot_coverage --@rules_rust//:extra_exec_rustc_flag='--cfg=feature="ot_coverage_enabled"'
+coverage:ot_coverage --//rules/coverage:enabled_flag
+coverage:ot_coverage --copt='-DOT_COVERAGE_ENABLED=1'
+
+# Workaround required for host-side unit tests
```

Comment:
What was the problem that required a "workaround"? can you add more details here like was there before?